PrepatcherCodeGenerator/PlayerDataAccessGenerator.cs

```diff
@@ -0,0 +1,132 @@
+
+[Generator]
+public class PlayerDataAccessGenerator : ISourceGenerator
+{
+    private bool IsValidField(IFieldSymbol symbol)
+    {
+        if (symbol.IsStatic || symbol.IsReadOnly)
+        {
+            return false;
+        }
+
+        if (symbol.DeclaredAccessibility != Accessibility.Public)
+        {
+            return false;
+        }
+        // We assume that the publicizer doesn't make things non-public, so we don't need to check the attribute param
+        if (symbol.GetAttributes().Any(attr => attr.AttributeClass?.ToDisplayString() == "BepInEx.AssemblyPublicizer.OriginalAttributesAttribute"))
+        {
+            return false;
+        }
+        // We would like to skip NonSerialized fields but the symbol.GetAttributes() doesn't seem to include NonSerialized
+
+        return true;
+    }
+
+    private void GetAccessorStrings(IFieldSymbol symbol, out string typeName, out string getAccessor, out string setAccessor)
+    {
+        if (symbol.Type.SpecialType == SpecialType.System_String)
+        {
+            typeName = "string";
+            getAccessor = "GetString";
+            setAccessor = "SetString";
+            return;
+        }
+        else if (symbol.Type.SpecialType == SpecialType.System_Boolean)
+        {
+            typeName = "bool";
+            getAccessor = "GetBool";
+            setAccessor = "SetBool";
+            return;
+        }
+        else if (symbol.Type.SpecialType == SpecialType.System_Single)
+        {
+            typeName = "float";
+            getAccessor = "GetFloat";
+            setAccessor = "SetFloat";
+            return;
+        }
+        else if (symbol.Type.SpecialType == SpecialType.System_Int32)
+        {
+            typeName = "int";
+            getAccessor = "GetInt";
+            setAccessor = "SetInt";
+            return;
+        }
+        else if (symbol.Type.Name == "Vector3")
+        {
+            typeName = "Vector3";
+            getAccessor = "GetVector3";
+            setAccessor = "SetVector3";
+            return;
+        }
+
+        typeName = symbol.Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+        getAccessor = $"GetVariable<{typeName}>";
+        setAccessor = $"SetVariable<{typeName}>";
+    }
+
+    public void Execute(GeneratorExecutionContext context)
+    {
+        Compilation compilation = context.Compilation;
+        INamespaceOrTypeSymbol playerDataSymbol = compilation.GlobalNamespace.GetMembers("PlayerData")
+            .FirstOrDefault(t => t.Kind == SymbolKind.NamedType && t.DeclaredAccessibility == Accessibility.Public);
+
+        if (playerDataSymbol == null)
+        {
+            throw new Exception("Could not find PlayerData");
+        }
+
+        List<IFieldSymbol> fieldsToDelegate = playerDataSymbol.GetMembers()
+            .OfType<IFieldSymbol>()
+            .Where(IsValidField)
+            .ToList();
+
+        StringBuilder sb = new();
+        sb.Append($@"
```

Comment:
I'd recommend triple-quote strings instead so you can indent them more naturally; it will strip the indentation based on the indentation of the line with the closing triple quote. See e.g. https://github.com/BadMagic100/Archipelago.MultiClient.Net.Analyzers/blob/master/Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Generators/DataStorageAttributeGenerator.cs